ops/lib.rs

```diff
@@ -721,42 +721,54 @@ fn codegen_arg(
     };
   }
   // Fast path for &/&mut [u8] and &/&mut [u32]
-  if let Some(ty) = is_ref_slice(&**ty) {
-    let (ptr_ty, mutability) = match ty {
-      SliceType::U8 => (quote!([u8]), quote!(&)),
-      SliceType::U8Mut => (quote!([u8]), quote!(&mut)),
-    };
-    return quote! {
-      let #ident = {
-        let value = args.get(#idx as i32);
-        if let Ok(view) = #core::v8::Local::<#core::v8::ArrayBufferView>::try_from(value) {
+  match is_ref_slice(&**ty) {
+    None => {}
+    Some(SliceType::U32Mut) => {
+      return quote! {
+        let #ident = if let Ok(view) = #core::v8::Local::<#core::v8::Uint32Array>::try_from(args.get(#idx as i32)) {
           let (offset, len) = (view.byte_offset(), view.byte_length());
           let buffer = match view.buffer(scope) {
               Some(v) => v,
               None => {
-                return #core::_ops::throw_type_error(scope, format!("Expected ArrayBufferView at position {}", #idx));
+                return #core::_ops::throw_type_error(scope, format!("Expected Uint32Array at position {}", #idx));
               }
           };
-          let store = buffer.get_backing_store();
-          if store.is_shared() {
-            return #core::_ops::throw_type_error(scope, format!("Expected non-shared ArrayBufferView at position {}", #idx));
-          }
-          unsafe { #mutability *(&store[offset..offset + len] as *const _ as *mut #ptr_ty) }
+          let store = buffer.data() as *mut u8;
+          // SAFETY: buffer from Uint32Array. Rust guarantees that lifetime of slice is no longer than the call.
+          unsafe { ::std::slice::from_raw_parts_mut(store.add(offset) as *mut u32, len / 4) }
         } else {
-          let b: #core::v8::Local<#core::v8::ArrayBuffer> = match value.try_into() {
-            Ok(v) => v,
+          return #core::_ops::throw_type_error(scope, format!("Expected Uint32Array at position {}", #idx));
+        };
+      };
+    }
+    Some(_) => {
+      return quote! {
```

Comment:
Nitpick: consider splitting this function into two helper functions (one for ArrayBuffer, other for Uint32Array). It becomes hard to right with that rightward drift.